VSTS/Extension/vss-extension.json

```diff
@@ -1,74 +1,131 @@
+                                                                     {
+                                                                         "name":  "SonarQube Projects",
+                                                                         "endpointUrl":  "{{endpoint.url}}api/projects/index",
+                                                                         "resultSelector":  "jsonpath:$[*]"
+                                                                     }
+                                                                 ],
+                                                 "authenticationSchemes":  [
+                                                                               {
+                                                                                   "type":  "ms.vss-endpoint.endpoint-auth-scheme-basic",
+                                                                                   "inputDescriptors":  [
+                                                                                                            {
+                                                                                                                "id":  "username",
+                                                                                                                "name":  "Token",
+                                                                                                                "description":  "Authentication Token generated by SonarQube.",
+                                                                                                                "inputMode":  "textbox",
+                                                                                                                "validation":  {
+                                                                                                                                   "isRequired":  true,
+                                                                                                                                   "dataType":  "string"
+                                                                                                                               }
+                                                                                                            },
+                                                                                                            {
+                                                                                                                "id":  "password",
+                                                                                                                "name":  "Password",
+                                                                                                                "description":  "Password",
+                                                                                                                "inputMode":  "passwordbox",

```

Comment:
Cool! I have't looked into custom endpoints, so I can't provide in depth feedback, but the one thing to test is how good the password masking is. With generic endpoints, if the user sets the username to "Foo" and the password to "Bar", whenever these strings are outputed to the logs, they are masked ( Foo -> ***,  helloFooWorld -. hello***World). As long as the same happens with your password, which is should, you're good to go.